### PR TITLE
server: migrate test-infra to testify for server/statistics_handler_test.go

### LINKED ISSUE #27807: migrate test-infra to testify for `server/statistics_handler_test.go`
This is a subtask of #26863

### PR DESCRIPTION
**Pull request by yedamao**
### What problem does this PR solve?

Issue Number: close #27807  

### Release note 

```release-note
None.
```

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -221,6 +222,49 @@ func (dbt *DBTest) mustQueryRows(query string, args ...interface{}) {
 	rows.Close()
 }
 
+// DBTestWithT like DBTest, but replace *check.C to *testing.T
+// TODO remove/rename when migrate server_test.go and statistics_handler_test.go to testify done
+type DBTestWithT struct {
```
Comment (xhebox) [inline server/server_test.go:227]: I think it can be abstract to the testkit package. Also, I remembered that there are https://github.com/jmoiron/sqlx.

**Review by xhebox — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks for your contribution @yedamao !

I don't think we should keep all structure previous test has. It is just one test case so we can write the code fall through directly. I'll push a commit with the comment by @xhebox

**Review by tisonkun — APPROVED** [review]
@xhebox please give this PR another pass. I make some non trivial changes based on your comment.

File: server/statistics_handler_serial_test.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	require.NotNil(t, fp1)
+	defer func() {
+		require.NoError(t, fp1.Close())
+		require.NoError(t, os.Remove(path1))
+	}()
+
+	resp2, err := client.fetchStatus("/stats/dump/tidb/test/" + snapshot)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, resp2.Body.Close())
```
Comment (xhebox) [inline server/statistics_handler_serial_test.go:115]: The most important reason why there are `terror.Call` to be lazy. :rofl:

**Review by xhebox — APPROVED** [review]

**Pull request merged**